File: src/equiv/AddDigits1.ml
module JB = Javalib_pack.JBasics
module QID = QualifiedIdentity
module E = Expr

let _ =
  let x = E.Var (Var.Mk (QID.QID ["x"], Type.Int)) in

  Equiv.run "AddDigits1"
            [(JB.TBasic `Int)]
            (JB.TBasic `Int)
            "addDigits1"
            "addDigits2"
            [x]
            [E.mk_igt x (E.Int 0)]

File: benchmark/equiv/AddDigits1.java
public class AddDigits1 {
  static public int addDigits1(int num) {
    int result = num - 9 * ((num - 1) / 9);
    return result;
  }

  static public int addDigits2(int num) {
    int result = num % 9 == 0 ? 9 : (num % 9);
    return result;
  }
}
